gradle/shared-with-buildSrc/code-quality-configuration.gradle.kts

```diff
@@ -71,6 +79,12 @@ fun Project.configureCodenarc(codeQualityConfigDir: File) {
                 }
             }
         }
+        val ruleClass = getIntegrationTestFixturesRule()
+        if (ruleClass != null) {
+            "codenarc"(files(ruleClass.protectionDomain!!.codeSource!!.location))
+            "codenarc"(embeddedKotlin("stdlib"))
+            "codenarc"(embeddedKotlin("runtime"))
+        }
```

Comment:
This is sort of gross.

I wonder if it would be better to just push the codenarc rule into the gradle/gradle build (and out of buildSrc) like a test fixture and only enforce this rule in the gradle/gradle build.  I think we'd use TestKit for integration tests in buildSrc, so this rule wouldn't need to be used in buildSrc.

Reply:
The thing I don't like about this is that we'd have a project in `gradle/gradle` that is *not* a "Gradle project". This is probably no big deal if we can reorganize the build to have different levels:

subprojects/tools/...
subproject/plugins/...

But I wanted a faster way to get this in.

Reply:
Right, but test fixtures are already not Gradle projects.  We have extensions for Spock too.

I'm not opposed to getting things in that will make things better.  This is just another kind of debt we'll have to fix later.

Reply:
I guess so, but this is not even test fixtures or Spock extensions, which need to be built before we execute the test. That's quality rules, so it really belongs to the build logic.